src/core/server/saved_objects/migrations/types.ts

```diff
@@ -22,13 +22,39 @@ import { SavedObjectsMigrationLogger } from './core/migration_logger';
 
 /**
  * A migration function defined for a {@link SavedObjectsType | saved objects type}
- * used to migrate it's {@link SavedObjectUnsanitizedDoc | documents}
+ * used to migrate it's {@link SavedObjectUnsanitizedDoc | documents} to a given version
+ *
+ * @example
+ * ```typescript
+ * const migrateProperty: SavedObjectMigrationFn = (doc, { log }) => {
+ *   try {
+ *     doc.attributes.someProp = migrateProperty(doc.attributes.someProp);
+ *   } catch(e) {
+ *     log.warn('Error migrating `someProp`');
```

Comment:
Catching migration exceptions is a common pattern in existing migrations but I think it's something we should discourage. It effectively causes a bug in some unknown place higher up in the stack when this object is eventually used. Also when we introduce "invalid" saved objects we want the migration system to catch and handle these migration exceptions.

Reply:
It was mostly to demonstrate a use of the `log` context property. I will remove the try/catch and add another dummy check instead